### ISSUE TITLE
Could not pass event PlayerInteractEvent to LogBlock v1.92 org.bukkit.event.EventException

### ISSUE THREAD

**Issue by mibby** [LogBlock/LogBlock#583]
LogBlock dev 234 - http://ci.md-5.net/job/LogBlock/234/
Spigot git-Spigot-35348a5-ee6d0fa (Latest compile as of 02/22)

```
[10:59:09] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to LogBlock v1.92
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:305) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:226) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PlayerInteractManager.interact(PlayerInteractManager.java:463) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PlayerConnection.a(PlayerConnection.java:724) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:50) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:80) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketHandleTask.run(SourceFile:13) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_31]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_31]
	at net.minecraft.server.v1_8_R1.MinecraftServer.z(MinecraftServer.java:696) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.DedicatedServer.z(DedicatedServer.java:316) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.MinecraftServer.y(MinecraftServer.java:634) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.MinecraftServer.run(MinecraftServer.java:537) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_31]
Caused by: java.lang.IllegalStateException: Start block missed in BlockIterator
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:170) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:233) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:268) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.entity.CraftLivingEntity.getLineOfSight(CraftLivingEntity.java:134) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.entity.CraftLivingEntity.getTargetBlock(CraftLivingEntity.java:190) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at de.diddiz.LogBlock.listeners.BlockBurnLogging.onExtinguish(BlockBurnLogging.java:38) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor189.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_31]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_31]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	... 16 more
[10:59:11] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to LogBlock v1.92
org.bukkit.event.EventException
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:305) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:502) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:487) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:226) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PlayerInteractManager.interact(PlayerInteractManager.java:463) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PlayerConnection.a(PlayerConnection.java:724) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:50) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketPlayInBlockPlace.a(PacketPlayInBlockPlace.java:80) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.PacketHandleTask.run(SourceFile:13) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_31]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_31]
	at net.minecraft.server.v1_8_R1.MinecraftServer.z(MinecraftServer.java:696) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.DedicatedServer.z(DedicatedServer.java:316) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.MinecraftServer.y(MinecraftServer.java:634) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at net.minecraft.server.v1_8_R1.MinecraftServer.run(MinecraftServer.java:537) [spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at java.lang.Thread.run(Thread.java:745) [?:1.8.0_31]
Caused by: java.lang.IllegalStateException: Start block missed in BlockIterator
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:170) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:233) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.util.BlockIterator.<init>(BlockIterator.java:268) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.entity.CraftLivingEntity.getLineOfSight(CraftLivingEntity.java:134) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at org.bukkit.craftbukkit.v1_8_R1.entity.CraftLivingEntity.getTargetBlock(CraftLivingEntity.java:190) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	at de.diddiz.LogBlock.listeners.BlockBurnLogging.onExtinguish(BlockBurnLogging.java:38) ~[?:?]
	at sun.reflect.GeneratedMethodAccessor189.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_31]
	at java.lang.reflect.Method.invoke(Method.java:483) ~[?:1.8.0_31]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:301) ~[spigot.jar:git-Spigot-35348a5-ee6d0fa]
	... 16 more
```

**Comment by frymaster** [comment]
The *good* news is we never used to even try to log this, so you aren't *losing* information, per se...

Do you know exactly what was happening when this error occurred?

@Mahagon any ideas? Is there a reason you used ``player.getTargetBlock()`` instead of ``event.getClickedBlock()`` anyway?

**Comment by Mahagon** [comment]
Hm i think i tried event.getClickedBlock() and nothing happened.

im at work atm, i will check that when im home (in ~6 hours)

**Comment by mibby** [comment]
Unfortunately I do not know what exactly was happening when the error occurred. Just skimmed through the server logs and came across this error. 

I don't believe someone has fire above Y 256.

**Comment by frymaster** [comment]
@Mahagon - don't forget, it only knows if it's fire or not **after** that line - at this point, it's firing on all ``PlayerInteractEvent``s

**Comment by frymaster** [comment]
@Mahagon - I've checked out some other fire-tracking code, and it looks like they fire would be the block **above** the one returned by ``event.getClickedBlock()``

of course, that has its own complications - I have no idea what happens when the side of a block is on fire, but I just realised I didn't test that with this code either...

My *guess* is that your code is more accurate, and I also guess that any time the above exception is called, it's not going to be a valid extinguish anyway (in which case we could catch and forget) but it'll need tested

**Comment by Mahagon** [comment]
@frymaster you are right, i think that was the problem. I used that code in another plugin and event.getClickedBlock was not reliable. http://git.wmchris.de/bdh/protectmyfireplace/blob/master/src/net/mahagon/ProtectMyFireplace/PlayerListener.java 
just thougt it would be nice for logging too ^^.

Adding the position and stuff to the debug will help a lot. I'll add that today.

**Comment by frymaster** [comment]
I note there's also a ``getBlockFace()`` method.  I *suspect* that if a combination of ``getClickedBlock() + getBlockFace = can tell where the fire block would be`` works, it'll be a less cpu-intensive solution than ``getTargetBlock``

**Comment by Mahagon** [comment]
@frymaster just created the pull request :)

b2005a0

**Comment by frymaster** [comment]
Fixed in 8c232b59970cc959a2a3fe10cf58f2e6ece86927
